### PR TITLE
Fixed error when finding gradient of function containing scalar-scalar multiply

### PR DESCRIPTION
**Pull request by alex-weaver**
For details of the error see issue 73: https://github.com/twitter/torch-autograd/issues/73

This error seems to appear whenever the function for which we want to find the gradient contains a scalar-scalar multiply. This affects any loss function with a weighting parameter (eg. the lambda parameter for L2 weight decay, or the beta parameter for inducing sparse activations with KL-divergence)

The example train-mnist-autoencoder.lua is affected by this since it makes use of L2 weight decay.

As far as I can see, the error is caused by the fact that operators.mul in gradfuns.lua makes the assumption that if A is not a tensor, then B is a tensor. This is not the case with scalar-scalar multiplication. Changing the condition to check that B is also a tensor means that in the scalar-scalar case it falls right through to the final else.

The pull request includes a unit test which exercises this case.

Edit: added slightly more detail.

### CODE REVIEW

**Comment by alexbw** [comment]
Thanks for the PR! Tests work for us. Merging (OpenBLAS is causing builds to fail on Travis, ignoring for now).
